plugins/handler/ceilometer-metrics/main.go

```diff
@@ -206,31 +177,48 @@ func genLabels(m ceilometer.Metric, publisher string, cNameShards []string) ([]s
 	labelKeys[1] = "publisher"
 	labelVals[1] = publisher
 
-	labelKeys[2] = "counter"
-	labelVals[2] = m.CounterName
-
 	var ctype string
 	if len(cNameShards) > 1 {
 		ctype = cNameShards[1]
 	} else {
 		ctype = cNameShards[0]
 	}
-	labelKeys[3] = "type"
-	labelVals[3] = ctype
 
-	labelKeys[4] = "project"
-	labelVals[4] = m.ProjectID
+	labelKeys[2] = "type"
+	labelVals[2] = ctype
 
-	labelKeys[5] = "unit"
-	labelVals[5] = m.CounterUnit
+	// non - critical
+	index := 3
```

Comment:
Hmm, I thought counter_* values are always present in ceilometer metric payload.

Reply:
You may be right, I did not attempt to confirm it